src/libraries/System.Private.CoreLib/src/System/Text/Ascii.Equality.cs

```diff
@@ -88,7 +110,10 @@ public static bool Equals(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
                 if (length % (uint)Vector512<TLeft>.Count != 0)
                 {
                     ref TLeft oneVectorAwayFromLeftEnd = ref Unsafe.Add(ref left, length - (uint)Vector512<TLeft>.Count);
-                    return TLoader.EqualAndAscii512(ref oneVectorAwayFromLeftEnd, ref oneVectorAwayFromRightEnd);
+                    if (!TLoader.EqualAndAscii512(ref oneVectorAwayFromLeftEnd, ref oneVectorAwayFromRightEnd))
+                    {
+                        return false;
```

Comment:
Why is this needed? It's the last elements, so that result can be returned (`true` / `false`) directly, w/o the need to jump the end of the method. Or does this result in better codegen? (Although I have to admin that with PGO it's now much harder for me to guess / expect the codegen here as the profile data can move the blocks around).

I'd revert these changes (same return as before) and let the JIT do it's thing.

Reply:
IIRC the JIT was adding extra return sections when I had it, but I can remove it for the simplicity